### PR TITLE
[THEMES-1374] Implemented logical properties for search-results list and results-list

### PR DESCRIPTION
**Pull request by malavikakoppula**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

Update padding, margin, and border properties. The following conversions must be done:

*-top and *-bottom need to be changed to *-block

*-left and *-right need to be changed to *-inline

height and width will need to be set to block-size and inline-size respectively

## Jira Ticket

- [THEMES-1374](https://arcpublishing.atlassian.net/browse/THEMES-1374)

## Acceptance Criteria

1. The above logical properties are applied and tested on the list blocks.
2. Storybook is updated with RTL.

## Test Steps

1. Checkout this branch `git checkout THEMES-1374`
3. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/search-results-list-block,@wpmedia/results-list-block`
4. Set "textDirection": "rtl" in the "siteProperties" object of the blocks.json
5. Check the styles for RTL matching to Figma designs

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

[THEMES-1374]: https://arcpublishing.atlassian.net/browse/THEMES-1374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: blocks/search-results-list-block/themes/news.json [hunk 1]
```diff
@@ -75,31 +78,34 @@
...
 						"display": "block",
-						"margin-top": "var(--global-spacing-4)"
+						"margin-block-start": "var(--global-spacing-4)"
 					},
 					"heading": {
 						"font-size": "var(--heading-level-5-font-size)",
 						"font-weight": "var(--heading-level-5-font-weight)",
-						"line-height": "var(--heading-level-5-line-height)",
-						"margin-bottom": "var(--global-spacing-4)"
+						"line-block-size": "var(--heading-level-5-line-height)",
```
Comment (vgalatro) [inline blocks/search-results-list-block/themes/news.json]: `line-height` is still the correct property to use. Could you revert all the places that was changed, thanks.

**Review by vgalatro — CHANGES_REQUESTED** [review]
This looks pretty good, just couple of issues I saw. First `line-height` does not need to be changed, as there is no `line-block-size` CSS property. Second could you add one story for each block that is RTL? You can check my other PRs for an example.

File: blocks/search-results-list-block/themes/news.json (hunk 1 above)
Comment (malavikakoppula) [reply blocks/search-results-list-block/themes/news.json]: @vgalatro reverted this [03f04dd](https://github.com/WPMedia/arc-themes-blocks/pull/1828/commits/03f04ddcc79cac1fad09c3b54fecc3c0856b705c)

**Comment by vgalatro** [comment]
@malavikakoppula , could you resolve the conflicts in the storybook style file, thanks.

File: .storybook/themes/news.scss [hunk 2]
```diff
@@ -2363,55 +2366,61 @@
...
 							"display": block,
-							"margin-top": var(--global-spacing-4)
+							"margin-block-start": var(--global-spacing-4)
 						),
 						"heading": (
 							"font-size": var(--heading-level-5-font-size),
 							"font-weight": var(--heading-level-5-font-weight),
-							"line-height": var(--heading-level-5-line-height),
-							"margin-bottom": var(--global-spacing-4)
+							"line-block-size": var(--heading-level-5-line-height),
```
Comment (vgalatro) [inline .storybook/themes/news.scss]: @malavikakoppula , `line-block-size` is still showing in the storybook `news.scss`, could you regenerate those styles?

**Review by vgalatro — APPROVED** [review]
LGTM!

**Pull request merged**